R/metapredict_merge.R

```diff
@@ -1,16 +1,29 @@
 makeMatchSampleMapping = function(metadata, subStudyNames, matchSampleColname) {
-  if (is.unsorted(subStudyNames)) {
-    arrangeFunc = function(x) dplyr::arrange(dplyr::desc(x))
-  } else {
-    arrangeFunc = dplyr::arrange}
-  metadataNow = metadata %>%
-    dplyr::filter(study %in% subStudyNames) %>%
-    dplyr::select(!!c('study', 'sample', matchSampleColname)) %>%
-    arrangeFunc(study)
-  mappingDf = metadataNow %>%
-    dplyr::group_by(!!matchSampleColname) %>%
-    dplyr::slice(1) %>%
-    dplyr::ungroup()
+
+  # if (is.unsorted(subStudyNames)) {
+  #   arrangeFunc = function(x) dplyr::arrange(dplyr::desc(x))
+  # } else {
+  #   arrangeFunc = dplyr::arrange}
+  # metadataNow = metadata %>%
+  #   dplyr::filter(study %in% subStudyNames) %>%
+  #   dplyr::select(!!c('study', 'sample', matchSampleColname)) %>%
+  #   arrangeFunc(study)
+
+  cols = c('study', 'sample')
+  if (!(matchSampleColname %in% cols)) {
+    cols = c(cols, matchSampleColname)}
+  metadataNow = data.table(metadata)[study %in% subStudyNames, ..cols]
+
+  order = if (is.unsorted(subStudyNames)) -1L else 1L
+  setorderv(metadataNow, 'study', order = order)
+
+  # mappingDf = metadataNow %>%
+  #   dplyr::group_by(!!matchSampleColname) %>%
+  #   dplyr::slice(1) %>%
+  #   dplyr::ungroup()
+
+  mappingDf = metadataNow[1,, by = ..matchSampleColname]
```

Comment:
Are you sure this does what you think? I would've done `metadataNow[, .SD[1L], by = ..matchSampleColname]`